esp/esp/program/modules/handlers/studentclassregmodule.py

```diff
@@ -430,19 +431,28 @@ def fillslot(self, request, tl, one, two, module, extra, prog):
                 classes = filter(lambda c: c.grade_min <=user_grade and c.grade_max >= user_grade, classes)
             classes = filter(lambda c: not c.isRegClosed(), classes)
 
-        #   Sort class list
-        classes = sorted(classes, key=lambda cls: cls.num_students() - cls.capacity)
-        classes = sorted(classes, key=lambda cls: cls.category.category)
-
         categories = {}
 
         for cls in classes:
-            categories[cls.category_id] = {'id':cls.category_id, 'category':cls.category_txt if hasattr(cls, 'category_txt') else cls.category.category}
+            categories[cls.category_id] = {'id':cls.category_id, 'category':cls.category_txt if hasattr(cls, 'category_txt') else cls.category.category, 'symbol':cls.category.symbol}
+
+        # Are the classes sorted by category? If so, by which aspect of category?
+        # Default is to sort by category symbol
+        catalog_sort = 'category__symbol'
```

Comment:
Can you factor out the logic in lines 441-450 into its own function to avoid duplicate code between fillslot and catalog_render?

Reply:
Done!